CMakeLists.txt

```diff
@@ -6,8 +6,9 @@ if(NOT CMAKE_C_STANDARD)
 endif()
 
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
Since we're going to change this anyway, i also suggests to set it one by one on each target that needed it, e.g.:

```cmake
add_library(${PROJECT_NAME} SHARED
  ...
)

set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
```

> Please note that only set it if a target needed it.